File: VB.NET/Controls/ReaderSettings/ParameterEditors/MaxThreadsCountEditorControl.vb

''' <summary>
''' Represents an editor control of ReaderSettings.MaximalThreadsCount parameter.
''' </summary>
Public Class MaxThreadsCountEditorControl
	Inherits ParameterEditorControl

	#Region "Constructors"

	Public Sub New()
		Minimum = 1
		Maximum = Math.Max(Environment.ProcessorCount * 2, BarcodeReaderSettings.MaximalThreadsCount)
		Value = BarcodeReaderSettings.MaximalThreadsCount
		Title = "Max threads count"
	End Sub

	#End Region



	#Region "Properties"

	''' <summary>
	''' Gets or sets a property value.
	''' </summary>
	Public Overrides Property Value() As Integer
		Get
			Return Math.Min(Maximum, Math.Max(Minimum, BarcodeReaderSettings.MaximalThreadsCount))
		End Get
		Set
			MyBase.Value = value
			BarcodeReaderSettings.MaximalThreadsCount = MyBase.Value
		End Set
	End Property

	#End Region

End Class

File: CSharp/Controls/ReaderSettings/ParameterEditors/MaxThreadsCountEditorControl.cs
using System;

namespace BarcodeDemo
{
    /// <summary>
    /// A control that allows to change the ReaderSettings.MaximumThreadCount parameter.
    /// </summary>
    public class MaxThreadCountEditorControl: ParameterEditorControl
    {

        #region Constructors

        /// <summary>
        /// Initializes a new instance of the <see cref="MaxThreadCountEditorControl"/> class.
        /// </summary>
        public MaxThreadCountEditorControl()
        {
            Minimum = 1;
            Maximum = Math.Max(Environment.ProcessorCount * 2, BarcodeReaderSettings.MaximumThreadCount);
            Value = BarcodeReaderSettings.MaximumThreadCount;
            Title = "Max threads count";
        }

        #endregion



        #region Properties

        /// <summary>
        /// Gets or sets a property value.
        /// </summary>
        public override int Value
        {
            get
            {
                return Math.Min(Maximum, Math.Max(Minimum, BarcodeReaderSettings.MaximumThreadCount));
            }
            set
            {
                base.Value = value;
                BarcodeReaderSettings.MaximumThreadCount = base.Value;
            }
        }

        #endregion

    }
}
